### DISCUSSION TITLE
Got Connection error

### DISCUSSION THREAD

**Discussion by casestudy12356** [sanny32/OpenModSim#23] [Q&A] [answered]
Master device tried to connect with Slave (Openmodsim), and sent a request for connect, but connection failed

Master logs 
 user.err Modbus: Error in connection with IP=192.168.122.148 and PORT=5020 for query[0] of a Group [1]
 
 TCP dump Slave PC:
 
 sudo tcpdump -i any -n -v port 5020
tcpdump: listening on any, link-type LINUX_SLL (Linux cooked v1), capture size 262144 bytes
17:08:16.468904 IP (tos 0x0, ttl 64, id 58468, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.122.1.60794 > 192.168.122.148.5020: Flags [S], cksum 0xe6f6 (correct), seq 1967781345, win 29200, options [mss 1460,sackOK,TS val 3815647588 ecr 0,nop,wscale 4], length 0
17:08:16.469815 IP (tos 0x0, ttl 64, id 38271, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.122.1.60796 > 192.168.122.148.5020: Flags [S], cksum 0xe6f0 (correct), seq 1967781348, win 29200, options [mss 1460,sackOK,TS val 3815647589 ecr 0,nop,wscale 4], length 0
17:08:16.471072 IP (tos 0x0, ttl 64, id 21846, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.122.1.60798 > 192.168.122.148.5020: Flags [S], cksum 0xe6ea (correct), seq 1967781351, win 29200, options [mss 1460,sackOK,TS val 3815647590 ecr 0,nop,wscale 4], length 0
17:08:16.511624 IP (tos 0x0, ttl 64, id 14703, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.122.1.60800 > 192.168.122.148.5020: Flags [S], cksum 0x92de (correct), seq 2654937299, win 29200, options [mss 1460,sackOK,TS val 3815647631 ecr 0,nop,wscale 4], length 0
17:08:16.516286 IP (tos 0x0, ttl 64, id 53383, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.122.1.60802 > 192.168.122.148.5020: Flags [S], cksum 0x92d5 (correct), seq 2654937302, win 29200, options [mss 1460,sackOK,TS val 3815647635 ecr 0,nop,wscale 4], length 0
17:08:16.520035 IP (tos 0x0, ttl 64, id 55511, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.122.1.60804 > 192.168.122.148.5020: Flags [S], cksum 0x92cc (correct), seq 2654937305, win 29200, options [mss 1460,sackOK,TS val 3815647639 ecr 0,nop,wscale 4], length 0
17:08:16.554754 IP (tos 0x0, ttl 64, id 54207, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.122.1.60806 > 192.168.122.148.5020: Flags [S], cksum 0xd209 (correct), seq 3130311969, win 29200, options [mss 1460,sackOK,TS val 3815647674 ecr 0,nop,wscale 4], length 0
17:08:17.595467 IP (tos 0x0, ttl 64, id 54208, offset 0, flags [DF], proto TCP (6), length 60)

sudo netstat -tulnp | grep :5020
tcp        0      0 192.168.122.148:5020    0.0.0.0:*               LISTEN      1234794/omodsim     

 sudo ss -tulnp | grep :5020
tcp   LISTEN 0       50                                   192.168.122.148:5020                                       0.0.0.0:*                                   users:(("omodsim",pid=1234794,fd=29))
                                          
:~$ sudo lsof -i :5020
COMMAND     PID      USER   FD   TYPE    DEVICE SIZE/OFF NODE NAME
omodsim 1234794 XX 29u  IPv4 312703125      0t0  TCP XX-00011:5020 (LISTEN)

 ps aux | grep omodsim
XX+ 1234790  0.0  0.0   2616   600 ?        S    15:58   0:00 /bin/sh /opt/OpenModSim/omodsim.sh
XX+ 1234794  0.2  0.4 1398732 130672 ?      Sl   15:58   0:13 /opt/OpenModSim/bin/omodsim
XX+ 1238601  0.0  0.0   8908   656 pts/2    S+   17:18   0:00 grep --color=auto omodsim

![image (1)](https://github.com/user-attachments/assets/6ff6d0c1-071c-4e6d-aa55-aa92de6ab99f)

Any Suggestions to resolve it

**Comment by sanny32** [comment]
Hi @casestudy12356. Can you try another modbus master client f.e. Open ModScan?

**Comment by casestudy12356** [comment, marked as answer]
It works only with modsim, and the firewall on the Linux machine needs to be disabled.
issue resolved.

thank you.
